internal/ethapi/util.go

```diff
@@ -4,14 +4,41 @@ import (
 	"context"
 	"math/big"
 
+	"github.com/celo-org/celo-blockchain/common"
 	"github.com/celo-org/celo-blockchain/contract_comm/currency"
+	"github.com/celo-org/celo-blockchain/core/types"
 	"github.com/celo-org/celo-blockchain/params"
 	"github.com/celo-org/celo-blockchain/rpc"
 )
 
-// NewCurrencyManager creates and returns a currencyManager pointing to the latest block
+func checkCeloTxFeeTx(ctx context.Context, b Backend, tx *types.Transaction) error {
```

Comment:
Imho I think that we can improve a little the naming here.
wdyt about `checkFeeFromTx` ?

Reply:
there's another function called checkFeeTx. I wanted to have `celo` somewhere in the name to make the difference clear

Reply:
`checkFeeFromCeloTx` maybe?